packages/renderer/src/lib/engine/Prune.svelte

```diff
@@ -1,27 +1,33 @@
 <script lang="ts">
-import Modal from '../dialogs/Modal.svelte';
 import type { EngineInfoUI } from './EngineInfoUI';
+import type { MessageBoxReturnValue } from '../../../../main/src/plugin/message-box';
 
 // Imported type for prune (containers, images, pods, volumes)
 export let type: string;
 
 // List of engines that the prune will work on
 export let engines: EngineInfoUI[];
 
-// Functionality for modal
-let openChoiceModal = false;
-function toggleChoiceModal(): void {
-  openChoiceModal = !openChoiceModal;
-}
-function keydownChoice(e: KeyboardEvent) {
-  e.stopPropagation();
-  if (e.key === 'Escape') {
-    toggleChoiceModal();
+async function openPruneDialog(): Promise<void> {
+  let message = 'This action will prune all unused ' + type;
+  if (engines.length > 1) {
+    message += ' from all container engines.';
+  } else {
+    message += ' from the ' + engines[0].name + ' engine.';
+  }
+
+  const result = await window.showMessageBox({
```

Comment:
I'm wondering we should not align with the extension API that exposes `showInformationMessage`, `showWarningMessage` and `showErrorMessage` rather that `showMessageBox` which is lower level

Reply:
The message box API is slightly lower level, but it only requires one method instead of 4 variants (info/warn/error/question) and using properties is much more extendable. e.g. we could continue to add properties similar to Electron API (https://www.electronjs.org/docs/latest/api/dialog) and allow detailed messages, checkbox, assign default/cancel buttons, etc. For instance our old telemetry dialog would have required message box b/c it had a checkbox.

If you'd still prefer I can just expose a showQuestionBox (or all 4 variants) for now.

Reply:
Yes you convinced me your choose the right path but maybe we should extend the extension API to have the same level.